apps/aevm/lib/gas.ex

```diff
@@ -0,0 +1,157 @@
+defmodule Gas do
+  @moduledoc """
+    Module for calculating the gas of a contract
+  """
+
+  use Bitwise
+
+  require OpCodesUtil
+  require GasCodes
+
+  @spec update_gas(integer(), map()) :: map() | {:error, String.t(), map()}
+  def update_gas(gas_cost, state) do
+    curr_gas = State.gas(state)
+
+    if curr_gas >= gas_cost do
+      gas_after = curr_gas - gas_cost
+      State.set_gas(gas_after, state)
+    else
+      throw({:error, "out_of_gas", state})
```

Comment:
I think we generally prefer to not to use `throw` in our code, but I understand that it's very difficult to avoid it.

Reply:
Yeah it is kinda difficult, we use the thrown state/message in the EVM tests because some of them fail on purpose and we have to catch the error as to not be considered an actual test fail